haystack/document_store/faiss.py

```diff
@@ -174,39 +174,37 @@ def update_embeddings(self, retriever: BaseRetriever, index: Optional[str] = Non
         self.faiss_index.reset()
 
         index = index or self.index
-        documents = self.get_all_documents(index=index)
 
-        if len(documents) == 0:
-            logger.warning("Calling DocumentStore.update_embeddings() on an empty index")
-            return
+        document_count = self.get_document_count(index=index)
```

Comment:
Not related to this PR how we will handle simultaneous/concurrent call to `update_embeddings` and `write_documents` functions. That will affect value of `self.get_document_count(index=index)` function and also create problem in `vector_id` duplication. Mainly do we need Global Lock as this looks like critical section.

Reply:
Yes, we'd need some guards for concurrent use of the `FAISSDocumentStore`. In the next weeks, we also plan to explore if there's a community interest in adding a document store like Milvus that already would take care of many scenarios pertaining to production deployments.